pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -413,94 +420,97 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
-
-		objectMap := stork_api.CreateObjectsMap(backup.Spec.IncludeResources)
-		info := stork_api.ObjectInfo{
-			GroupVersionKind: metav1.GroupVersionKind{
-				Group:   "core",
-				Version: "v1",
-				Kind:    "PersistentVolumeClaim",
-			},
-		}
+	pvcMappings := make(map[string][]v1.PersistentVolumeClaim)
+	backup.Status.Stage = stork_api.ApplicationBackupStageVolumes
+	backup.Status.Volumes = make([]*stork_api.ApplicationBackupVolumeInfo, 0)
 
-		for _, namespace := range backup.Spec.Namespaces {
-			pvcList, err := core.Instance().GetPersistentVolumeClaims(namespace, backup.Spec.Selectors)
-			if err != nil {
-				return fmt.Errorf("error getting list of volumes to migrate: %v", err)
-			}
+	objectMap := stork_api.CreateObjectsMap(backup.Spec.IncludeResources)
+	info := stork_api.ObjectInfo{
+		GroupVersionKind: metav1.GroupVersionKind{
+			Group:   "core",
+			Version: "v1",
+			Kind:    "PersistentVolumeClaim",
+		},
+	}
 
-			for _, pvc := range pvcList.Items {
-				// If a list of resources was specified during backup check if
-				// this PVC was included
-				info.Name = pvc.Name
-				info.Namespace = pvc.Namespace
-				if len(objectMap) != 0 {
-					if val, present := objectMap[info]; !present || !val {
-						continue
-					}
-				}
+	for _, namespace := range backup.Spec.Namespaces {
+		pvcList, err := core.Instance().GetPersistentVolumeClaims(namespace, backup.Spec.Selectors)
+		if err != nil {
+			return fmt.Errorf("error getting list of volumes to migrate: %v", err)
+		}
 
-				// Don't backup pending or deleting PVCs
-				if pvc.Status.Phase != v1.ClaimBound || pvc.DeletionTimestamp != nil {
+		for _, pvc := range pvcList.Items {
+			// If a list of resources was specified during backup check if
+			// this PVC was included
+			info.Name = pvc.Name
+			info.Namespace = pvc.Namespace
+			if len(objectMap) != 0 {
+				if val, present := objectMap[info]; !present || !val {
 					continue
 				}
-				driverName, err := volume.GetPVCDriver(&pvc)
-				if err != nil {
-					// Skip unsupported PVCs
-					if _, ok := err.(*errors.ErrNotSupported); ok {
-						continue
-					}
-					return err
+			}
+
+			// Don't backup pending or deleting PVCs
+			if pvc.Status.Phase != v1.ClaimBound || pvc.DeletionTimestamp != nil {
+				continue
+			}
+			driverName, err := volume.GetPVCDriver(&pvc)
+			if err != nil {
+				// Skip unsupported PVCs
+				if _, ok := err.(*errors.ErrNotSupported); ok {
+					continue
 				}
-				if driverName != "" {
-					if pvcMappings[driverName] == nil {
-						pvcMappings[driverName] = make([]v1.PersistentVolumeClaim, 0)
-					}
-					pvcMappings[driverName] = append(pvcMappings[driverName], pvc)
+				return err
+			}
+			if driverName != "" {
+				if pvcMappings[driverName] == nil {
+					pvcMappings[driverName] = make([]v1.PersistentVolumeClaim, 0)
 				}
+				pvcMappings[driverName] = append(pvcMappings[driverName], pvc)
 			}
 		}
+	}
+
+	if len(backup.Status.Volumes) != len(pvcMappings) {
```

Comment:
This will always be true since `backup.Status.Volumes` is being initialized on line 425

Reply:
Fixed the issue.

**Manually Tested below use case:**
1) In the code, made the batch count to one (testing purpose) and added a sleep of 5 mints before updating the CR content after calling driver.StartBackup().
2) After sleep, I was manually updating the CR content directly using "kubectl edit", To make the CR update fail from the stork code.
3) After starting first volume backup, when it was waiting (sleep), I manually updated the CR. This made the CR update fail and returned.
4) When the reconciler was called again, It started from the beginning.
5) This time, I allowed the first volume update to go through and during the wait for second volume backup, I manually updated CR again.
6) The update failed again and this time, the backup.Volume.Status was not nil and len(backup.Status.Volumes) != pvcCount. 
7) So it continued again from the beginning and also the make was called again. 
8) Then allowed the backup to continue and it completed successfully.

**Also ran the TestApplicationBackup focus integration test with the change.**
https://jenkins.portworx.dev/job/Stork/job/stork-2-dev-test/103/